src/components/RandomProductsBanner/RandomProductsBanner.js

```diff
@@ -1,9 +1,131 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { FormattedMessage } from 'react-intl';
+
+import LoadingSpinner from '../LoadingSpinner';
+import messages from './messages';
+import { fetchBanner } from '../../containers/Dashboard/actions';
+import {
+  makeSelectProducts,
+  makeSelectCntProducts,
+  makeSelectLoading,
+  makeSelectError,
+} from '../../containers/Dashboard/selectors';
+import { randomProductsBannerOptions } from './constants';
+
+import './RandomProductsBanner.css';
+
+const { updateInterval, cntRandomProducts } = randomProductsBannerOptions;
+
+const getProductItem = product => {
+  return (
+    <li
+      key={product.get('id')}
```

Comment:
Only the id can have duplicates ib future. Better to add some more data: `random-product-banner-item-${id}`